r/content/tables.Rmd

```diff
@@ -0,0 +1,251 @@
+This documentation lists all available compute functions, any associated options classes 
+they need, and the valid data types that they can be used with.
+
+You can list all available Arrow compute functions from R by calling 
+`list_compute_functions()`.
+
+```{r, list_compute_funcs}
+list_compute_functions()
+```
+```{r, test_list_compute_funcs, opts.label = "test"}
+test_that("list_compute_funcs", {
+  expect_gt(length(list_compute_functions()), 0)
+})
+```
+
+The majority of functions here have been mapped to their base R or tidyverse 
+equivalent and can be called within a dplyr query as usual.  For functions which
+don't have a base R or tidyverse equivalent, or you want to supply custom 
+options, you can call them by prefixing their name with "arrow_".  
+
+For example, base R's `is.na()` function is the equivalent of the Arrow C++ 
+compute function `is_null()` with the option `nan_is_null` set to `TRUE`.  
+A mapping between these functions (with `nan_is_null` set to `TRUE`) has been
+created in arrow.
+
+```{r, dplyr_is_na}
+demo_df <- data.frame(x = c(1, 2, 3, NA, NaN))
+
+demo_df %>%
+  Table$create() %>%
+  mutate(y = is.na(x)) %>% 
+  collect()
+```
+
+```{r, test_dplyr_is_na, opts.label = "test"}
+test_that("dplyr_is_na", {
+  out <- demo_df %>%
+  Table$create() %>%
+  mutate(y = is.na(x)) %>% 
+  collect()
+  
+  expect_equal(out$y, c(FALSE, FALSE, FALSE, TRUE, TRUE))
+  
+})
+```
+
+If you want to call Arrow's `is_null()` function but with `nan_is_null` set to 
+`FALSE` (so it returns `TRUE` when a value being examined is `NA` but `FALSE` 
+when the value being examined is `NaN`), you must call `is_null()` directly and 
+specify the option `nan_is_null = FALSE`.
+
+```{r, dplyr_arrow_is_null}
+demo_df %>%
+  Table$create() %>%
+  mutate(y = arrow_is_null(x, options  = list(nan_is_null = FALSE))) %>% 
+  collect()
+```
+
+```{r, test_dplyr_arrow_is_null, opts.label = "test"}
+test_that("dplyr_arrow_is_null", {
+  out <- demo_df %>%
+    Table$create() %>%
+    mutate(y = arrow_is_null(x, options  = list(nan_is_null = FALSE))) %>% 
+    collect()
+  
+  expect_equal(out$y, c(FALSE, FALSE, FALSE, TRUE, FALSE))
+  
+})
+```
+
+#### Compute functions with options
+
+Although not all Arrow C++ compute functions require options to be specified, 
+most do, and for these functions to work in R, the function must be associated 
+with the appropriate options C++ class in the R 
+package's C++ code.  At the time of writing, all compute functions available in
+the development version of the arrow R package had been associated with their options
+classes.  However, as the Arrow C++ library's functionality extends, compute 
+functions may be added which do not yet have an R binding.  If you find a C++ 
+compute function which you wish to use from the R package, please [open an issue
+on the project JIRA](https://issues.apache.org/jira/projects/ARROW/issues).
```

Comment:
As a user, I don't really know what "the function must be associated with the appropriate options C++ class in the R  package's C++ code" means.

I feel like the user is going to kind of gloss over this section.  What would a missing options error even look like?

Reply:
Have rephrased to clarify.  I don't mind if the reader glosses over this, as this section is only relevant to a very small niche of users.  

The current error messaging isn't quite there, so I opened this ticket  -  [ARROW-14067](https://issues.apache.org/jira/browse/ARROW-14067) - to improve the error messaging if options are not bound or not specified by the user.